File: Exercise PY/3-compareTheTripletes.py
# Jogo entre Bob e Alice 
# Nest jogo comparamos o arr de cada um, aquele que tiver um indice maior acrecentamos um ponto.
# A comparação seria da seguinte forma:
#   - a[0]>b[0] -> Ponto para alice
#   - a[1]<b[1] -> Ponto para bob
#   - a[2]=b[2] -> nao computamos pontos
#  Exemplo:
# a = [1, 2, 3]
# b = [3, 2, 1]
# resposta = [1,1]


def compareTheTriplete(a, b):
  aliceScore = 0;
  bobScore = 0;
  for alice, bob in zip(a, b):
    if alice > bob: aliceScore += 1;
    if alice < bob: bobScore += 1;

  return [aliceScore, bobScore];

a = [1, 2, 3]
b = [3, 2, 1]
result = compareTheTriplete(a, b);
print(result);


#ZIP combina elementos de várias sequências

File: Hackerrank/Exercise JS/3-compareTheTripletes.js
/*
  Exercicios exigem uma comparação entre indices de cada arr
  Em um jogo Bob e Alice querem descobrir quem pontua mais.
  A comparação seria da seguinte forma:
    - a[0]>b[0] -> Ponto para alice
    - a[1]<b[1] -> Ponto para bob
    - a[2]=b[2] -> nao computamos pontos

  Exemplo:
  a = [1, 2, 3]
  b = [3, 2, 1]
  resposta = [1,1]
*/


const compareTheTriplete = (a, b) =>{
  let aliceScore = 0;
  let bobScore = 0;

  for(let i = 0; i < 3; i++){
    if(a[i] > b[i]){
      aliceScore += 1;
    }
    if(a[i] < b[i]){
      bobScore += 1;
    }
  }

  return [aliceScore, bobScore];
}

const a = [1, 2, 3];
const b = [3, 2, 1];
const result = compareTheTriplete(a, b);
console.log(result);